src/languageclient.ts

```diff
@@ -0,0 +1,266 @@
+                            repo: 'terraform-lsp'
+                        });
+
+                        if (this.currentReleaseId !== release.data.id.toString()) {
+                            vscode.window.showInformationMessage(`A newer Lanaguage Server is available ${release.data.tag_name}. Run the 'Install/Update Language Server' command to update`);
+                        }
+                    } catch (e) {
+                        return vscode.window.showErrorMessage(`Failed checking for newer language server version. Error: ${e}`);
+                    }
+
+                } catch (e) {
+                    vscode.window.showErrorMessage(`Failed starting language server. Error: ${e}`);
+                    reject(e);
+                }
+            });
+        });
+
+    }
+
+    private async installCommonProviders(serverLocation: string) {
+        let terminal = vscode.window.createTerminal("Terraform: Install providers");
+        const defaultProvidersPath = Path.join(serverLocation, "providers.tf");
+        if (!fs.existsSync(defaultProvidersPath)) {
+            fs.writeFileSync(defaultProvidersPath, `
+provider "aws" {}
+provider "azurerm" {}
+provider "google" {}
+provider "alicloud" {}
+provider "helm" {}
+provider "kubernetes" {}
+provider "random" {}
+provider "null" {}
+provider "external" {}
+provider "template" {}
+provider "archive" {}
+                        `);
+        }
+        const defaultProvidersPluginsPath = Path.join(serverLocation, ".terraform", "plugins");
+        if (!fs.existsSync(defaultProvidersPluginsPath)) {
+            terminal.show();
+            terminal.sendText(`cd ${serverLocation} && terraform init`);
+            terminal.sendText(`echo "Done" && exit 0`);
+            let terminalPid = await terminal.processId;
+            // Hack to wait on the process to exit
+            let isRunning = true;
+            while (isRunning) {
+                try {
+                    ps.kill(terminalPid, 0);
+                } catch (e) {
+                    isRunning = false;
+                }
+            }
+        }
+        const providerBinaries = this.findFilesInDir(defaultProvidersPluginsPath, "terraform-provider");
+        providerBinaries.forEach(file => {
+            const newLocation = Path.join(serverLocation, file.filename);
+            fs.copyFileSync(file.fullPath, newLocation);
+            fs.chmodSync(newLocation, '777');
+        });
+    }
+
+    /**
+     * Find all files recursively in specific folder with specific extension, e.g:
+     * findFilesInDir('./project/src', '.html') ==> ['./project/src/a.html','./project/src/build/index.html']
+     * @param  {String} startPath    Path relative to this file or other file which requires this files
+     * @param  {String} filter       Extension name, e.g: '.html'
+     * @return {Array}               Result files with path string in an array
+     */
+    private findFilesInDir(startPath: string, filter: string) {
```

Comment:
isn't there some `glob()` function we can use from somewhere else? I thought we even require it already (maybe only for building and maybe only transitively)?

Reply:
Cool, out of practice with `node`, I've moved this over to use `glob` from `npm`. :)